### ISSUE TITLE
Need a 0.21.0 release?

### ISSUE THREAD

**Issue by thomasvl** [bazelbuild/buildtools#497]
Maybe whatever automation is used for bazel releases can be extended to ensure bazeltools also gets a release at the same time?

**Comment by thomasvl** [comment]
Bazel did a 0.22.0 release, and still no update here.

**Comment by laurentlb** [comment]
Release process is still manual, as far as I know.

**Comment by laurentlb** [comment]
Vlad, can you make a new release?

**Comment by vladmos** [comment]
@pmbethe09 Could you please tell me how the releases are made, so I can do them on my own?

**Comment by pmbethe09** [comment]
@vladmos -- I had been hoping for more progress on the automated build front with help from core bazel folks -- but nothing yet.

It is not very exciting -- assuming version 0.22.0:
* get the relevant bazel 0.22.0 -- I usually build from src
* in buildtools: git tag 0.22.0; git push tags origin 0.22.0
* in buildtools: git checkout 0.22.0; bazel build -c opt --workspace_status_command=$PWD/status.sh unused_deps buildifier buildozer

I do this both on linux and Mac, then copy the created binaries into the release which I create using the newly pushed tag.

I can get to this tomorrow unless you want to try.

**Comment by thomasvl** [comment]
Since that release got done (but things are behind again), should this be closed or morphed into getting a process in place to do releases with bazel releases?

**Comment by pmbethe09** [comment]
I have been looking for guidance from the core bazel folks on how to
automate a release process that includes Mac, win, linux

**Comment by thomasvl** [comment]
I'll also add that the buildifier support added to buildkite appears to use a version of buildifier that is not a release at all.  adding to the confusion of when it flags something as a version they get here can produce different results.

@pmbethe09 is there someone to loop in to get the clarity you need?

**Comment by laurentlb** [comment]
@fweikert should the CI use the latest release?

**Comment by vladmos** [comment]
A new release can contain newly added lints, updating buildifier on the CI automatically may cause new findings to block presubmits.

**Comment by fweikert** [comment]
In the long term we would like users to specify the desired Buildifier version in the CI presubmit configuration.

**Comment by philwo** [comment]
I don't really understand how the relation between Buildtools and Bazel is when it comes to versions. Should there be a 1:1 mapping for Buildtools releases and Bazel releases? If so, how do we know which exact commit in Buildtools is good for a given Bazel release (thus doesn't already contain lint / needed migration checks that are not yet relevant for a given Bazel release)?

If that's not the case and we just release Buildtools on its own whenever we feel like it (e.g. we could to weekly releases or every time CI is green or ...), then how do people know which version of Buildtools to use for their Bazel release?

If we want to do Buildtools releases together with Bazel, can we add the steps needed to do a buildtools release to the [Bazel release manager playbook](https://github.com/bazelbuild/continuous-integration/blob/master/docs/release-playbook.md) and then let the Bazel release manager do them whenever we do a Bazel release?

**Comment by thomasvl** [comment]
As a users of it, I can't say I care if the numbers match or not.  If I get a vote, I'd just cast one for regular updates.  Since buildifier is now the provider of linting for Starlark, getting timely updates to address improvements/bugs/etc. feels important.

**Comment by pmbethe09** [comment]
When I cut a release I pretty much just use whatever the latest bazel release is.  There is not much science going on.

**Comment by fweikert** [comment]
@pmbethe09: We now have a Buildkite pipeline that builds binaries on Mac, Windows and Linux: https://buildkite.com/bazel-trusted/buildtools-binaries

The release process is still mostly manual, but we're also planning to improve that (e.g. GitHub integration).

**Comment by pmbethe09** [comment]
Nice.  Letting @vladmos handle the releases for now, but I seem to have
permission in case I have to make one in the future.

*From: *Florian Weikert <notifications@github.com>
*Date: *Wed, May 15, 2019 at 12:43 PM
*To: *bazelbuild/buildtools
*Cc: *Paul Bethe, Mention

@pmbethe09 <https://github.com/pmbethe09>: We now have a Buildkite pipeline

**Comment by thomasvl** [comment]
@fweikert https://buildkite.com/bazel-trusted/buildtools-binaries comes back as "Page Not Found" for me.

**Comment by fweikert** [comment]
Yeah, the pipeline is currently only visible to the release managers (pmbethe, laurentlb and vladmos). I'm not sure whether we should make it public.
